src/olympia/search/filters.py

```diff
@@ -135,7 +137,23 @@ class AddonGuidQueryParam(AddonQueryParam):
     es_field = 'guid'
 
     def get_value(self):
-        value = self.request.GET.get(self.query_param)
+        value = self.request.GET.get(self.query_param, '')
+
+        # Hack for Firefox 'return to AMO' feature (which, sadly, does not use
+        # a specific API but rather encodes the guid and adds a prefix to it,
+        # only in the search API): if the guid param matches this format, and
+        # the feature is enabled through a setting, then we decode it and
+        # proceed. A 400 is returned if the format is recognized but the
+        # feature is disabled through the setting, acting as a kill-switch.
+        if value.startswith('rta:') and '@' not in value:
+            if settings.RETURN_TO_AMO is not True:
+                raise ValueError('RTA is currently disabled')
+            # Any ValueError will trigger a 400.
+            try:
+                value = urlsafe_base64_decode(value[4:])
```

Comment:
You could technically encoded a list of guids, comma-separated, because we're not just returning the value at this point.  Maybe we want this (though I'm not sure why) and maybe it's a bug report waiting to happen when a guid containing a comma that isn't appropriately encoded is sent by Firefox.  
In either case, the docs above say " a single guid can be passed in base64url format," which isn't entirely correct.